### PR TITLE
Submission onepage

### PR DESCRIPTION
**Pull request by LakshitXD**
Submissions-Onepage-ui for frontend

[Notion Task Link](https://www.notion.so/3d6d8bec59b7483bbaeb98bac21458ec?v=73a0d569661641dbb7e4c318e2b6118f&p=5a66c2ce725943a69cbfc3d7763555ae)

### CODE REVIEW

**Review by Adhikram — APPROVED** [review]

File: devsnest/app/controllers/api/v1/frontend_submissions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    # frontend submission controller
+    class FrontendSubmissionsController < ApplicationController
+      include JSONAPI::ActsAsResourceController
+      before_action :simple_auth, only: [:create]
+
+      def create
```
Comment (vedansh) [inline devsnest/app/controllers/api/v1/frontend_submissions_controller.rb:10]: can we just use the resource function for this?

File: devsnest/app/resources/api/v1/content_resource.rb [hunk 2]
```diff
@@ -18,23 +19,33 @@ def self.default_sort
       def questions_list
         contents = Content.where(id: video_questions)
         return [] if video_questions.nil?
+
         questions = []
         contents.each do |c|
+          links = FrontendSubmission.where(user_id: context[:user].id, content_id: c.id).first
```
Comment (vedansh) [inline devsnest/app/resources/api/v1/content_resource.rb:25]: it should be link, name it better

File: devsnest/app/resources/api/v1/content_resource.rb [hunk 3]
```diff
@@ -18,23 +19,33 @@ def self.default_sort
...
           sub = Submission.where(user_id: context[:user].id, content_id: c.id).first
 
-          questions.push c.as_json.merge(status: sub.present? ? sub.status: "notdone")
+          questions.push c.as_json.merge(status: sub.present? ? sub.status : 'notdone', submission_link: links.present? ? links.submission_link : 'null')
```
Comment (vedansh) [inline devsnest/app/resources/api/v1/content_resource.rb:28]: nil, never send 'null' as string

File: devsnest/app/controllers/api/v1/frontend_submissions_controller.rb (hunk 1 above)
Comment (LakshitXD) [reply devsnest/app/controllers/api/v1/frontend_submissions_controller.rb:10]: Can you clarify more i dont get it. Do you want me to create before create for this?

Comment (vedansh) [reply devsnest/app/controllers/api/v1/frontend_submissions_controller.rb:9]: don't have a create function, let it be handled by resource

Comment (LakshitXD) [reply devsnest/app/controllers/api/v1/frontend_submissions_controller.rb:9]: ok

Comment (LakshitXD) [reply devsnest/app/controllers/api/v1/frontend_submissions_controller.rb]: done

File: devsnest/app/resources/api/v1/content_resource.rb [hunk 4]
```diff
@@ -18,22 +19,32 @@ def self.default_sort
...
         contents = Content.where(id: video_questions)
         return [] if video_questions.nil?
+
         questions = []
         contents.each do |c|
+          link = FrontendSubmission.where(user_id: context[:user].id, content_id: c.id).first
           sub = Submission.where(user_id: context[:user].id, content_id: c.id).first
 
-          questions.push c.as_json.merge(status: sub.present? ? sub.status: "notdone")
+          questions.push c.as_json.merge(status: sub.present? ? sub.status : 'notdone', submission_link: link.present? ? link.submission_link : 'nil')
```
Comment (vedansh) [inline devsnest/app/resources/api/v1/content_resource.rb:28]: nil is not string @LakshitXD

File: devsnest/app/resources/api/v1/content_resource.rb [hunk 5]
```diff
@@ -18,22 +19,32 @@ def self.default_sort
...
-        return "notdone"
+
+        return 'notdone'
+      end
+
+      def submission_link
+        user_id = context[:user].id
+        submission = FrontendSubmission.where(user_id: user_id, content_id: @model.id)
+
+        return submission.first.submission_link if submission.present?
```
Comment (vedansh) [inline devsnest/app/resources/api/v1/content_resource.rb:47]: what happens if submission is not present?

Comment (LakshitXD) [reply devsnest/app/resources/api/v1/content_resource.rb:47]: it return null

**Pull request merged**